nir/src/main/scala/scala/scalanative/nir/Types.scala

```diff
@@ -88,18 +89,49 @@ object Type {
   final case class Var(ty: Type)                        extends SpecialKind
   final case class Function(args: Seq[Type], ret: Type) extends SpecialKind
 
-  val unbox = Map[Type, Type](
-    Type.Ref(Global.Top("java.lang.Boolean"))   -> Type.Bool,
-    Type.Ref(Global.Top("java.lang.Character")) -> Type.Char,
-    Type.Ref(Global.Top("java.lang.Byte"))      -> Type.Byte,
-    Type.Ref(Global.Top("java.lang.Short"))     -> Type.Short,
-    Type.Ref(Global.Top("java.lang.Integer"))   -> Type.Int,
-    Type.Ref(Global.Top("java.lang.Long"))      -> Type.Long,
-    Type.Ref(Global.Top("java.lang.Float"))     -> Type.Float,
-    Type.Ref(Global.Top("java.lang.Double"))    -> Type.Double
+  val boxesTo = Seq[(Type, Type)](
+    Type.Ref(Global.Top("scala.scalanative.native.Word"))   -> Type.Word,
+    Type.Ref(Global.Top("scala.scalanative.native.UWord"))  -> Type.Word,
```

Comment:
I'd reorder the two above. We put the "canonical" box for given primitive type last.